### PR TITLE
test: Add some Vue tests and a setup for them

### PR DESCRIPTION
**Pull request by salisdemos**
I restored the Vue tests that got lost during the great shift.

Now, if You run `yarn run test` not only the js files get tested, but some Vue files too.

I did not changed the tests itself.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,20 +1,66 @@
+/**
+ * (c) 2010-present DEMOS E-Partizipation GmbH.
+ *
+ * This file is part of the package demosplan,
+ * for more information see the license file.
+ *
+ * All rights reserved
+ */
+
```
Comment (hwiem) [inline jest.config.js:9]: I believe this block should be removed (here and other files)?

File: jest.config.js [hunk 2]
```diff
@@ -1,20 +1,66 @@
...
 /** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+const path = require('path')
+
+console.dir('<rootDir>')
```
Comment (hwiem) [inline jest.config.js:13]: (Why) Is this needed?

File: tests/__mocks__/dplan_api_map_options_admin.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (hwiem) [inline tests/__mocks__/dplan_api_map_options_admin.json:1]: Where is this used? (Wondering the same for some other files in the mocks directory)

File: tests/prefixClass.spec.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  it('should return a list of classes prefixed', () => {
+    expect(prefixClass('test foo bar')).toEqual('dp-test dp-foo dp-bar')
+  })
+
+  it('should return a quereyselector with prefixed classes', () => {
+    expect(prefixClass('#test .foo .bar')).toEqual('#test .dp-foo .dp-bar')
+    expect(prefixClass('[data-some-stuff="its all right"] > .a-class > label')).toEqual('[data-some-stuff="its all right"] > .dp-a-class > label')
+  })
+
+  it('should return the same sting if there are no points at the beginning of a word', () => {
```
Comment (hwiem) [inline tests/prefixClass.spec.js:28]:
```suggestion
  it('should return the same string if there are no points at the beginning of a word', () => {
```

**Review by hwiem — CHANGES_REQUESTED** [review]

File: tests/__mocks__/dplan_api_map_options_admin.json (hunk 3 above)
Comment (salisdemos) [reply tests/__mocks__/dplan_api_map_options_admin.json:1]: I can't tell. I just moved it. Cleaning up would be a next step.

File: jest.config.js (hunk 2 above)
Comment (salisdemos) [reply jest.config.js:4]: its not 4dcb065c73634144073bcb626a8e19d43b66cd07

File: jest/shallowMountWithGlobalMocks.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/**
```
Comment (hwiem) [inline jest/shallowMountWithGlobalMocks.js:1]: This block still needs to be removed

File: tests/__mocks__/dplan_api_map_options_admin.json (hunk 3 above)
Comment (hwiem) [reply tests/__mocks__/dplan_api_map_options_admin.json:1]: I think we should make sure we need the files before adding them (apart from that, I agree that we can clean up things later)

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,7 @@
 ### Added
 - ([#269](https://github.com/demos-europe/demosplan-ui/pull/269)) DpIcon: new selector `landscape`, `portrait`, or `square`, based on icon proportions ([@spiess-demos](https://github.com/spiess-demos))
 - ([#270](https://github.com/demos-europe/demosplan-ui/pull/270)) DpIcon: new icons (ai, copy) ([@spiess-demos](https://github.com/spiess-demos))
+- ([#268](https://github.com/demos-europe/demosplan-ui/pull/268)) Tests: Add Testsetup for Vue Files and some Tests ([@salis-demos](https://github.com/salis-demos))
```
Comment (hwiem) [inline CHANGELOG.md:8]: don't we add the newest entry at the top of the section?

**Review by hwiem — CHANGES_REQUESTED** [review]
The files `Filter.mock.js`, `layer_json.mock.js`, and `MapLayer.mock.js` are not used anywhere, so they should not be added to demosplan-ui

File: CHANGELOG.md (hunk 6 above)
Comment (salisdemos) [reply CHANGELOG.md]: do we? We can though. I haven't thought about it.

**Comment by salisdemos** [comment]
> The files `Filter.mock.js`, `layer_json.mock.js`, and `MapLayer.mock.js` are not used anywhere, so they should not be added to demosplan-ui

I don't know why i am so unconcentrated. Sorry!
Removed all the unused mocks 6afd8abf37d5559c161434ab39b5591834629071

File: CHANGELOG.md (hunk 6 above)
Comment (salisdemos) [reply CHANGELOG.md]: changed in 3edbde43e14a994be1aa5c2bca4282688c7b255c

**Review by hwiem — APPROVED** [review]
Thanks for adding the tests :heart:

**Pull request merged**